src/components/board-game/index.tsx

```diff
@@ -36,44 +39,50 @@ function Board() {
         if (isEqualCards(firstCard, secondCard)) {
             setActiveCards(() => []);
             setShuffleArr((prevCards) => prevCards.map(
-                (item) => isEqualIDSelectedCard(item, {firstCard, secondCard}) ? ({...item, stateCard: constants.INACTIVE}) : item)
+                (item) => isEqualIDSelectedCard(item, {firstCard, secondCard}) ? ({...item, stateCard: CARD_STATES.INACTIVE}) : item)
             );
         } else {
             closePairID.current = setTimeout(() => {
                 closePair();
                 closePairID.current = null;
             }, 1000);
 
-            return () => clearTimeout(closePairID.current);
+            return () => {
+                if(closePairID.current !== null) {
+                    return clearTimeout(closePairID.current);
+                }
+            }
         }
     }, [activeCards.length]);
 
     function closePair() {
         setShuffleArr((cards) => cards.map(
-            (card) => activeCards.some((activeCard) => activeCard.id === card.id) ? ({...card, isVisible: false, stateCard: constants.IN_PROGRESS}) : card
+            (card) => activeCards.some((activeCard) => activeCard.id === card.id) ? ({...card, isVisible: false, stateCard: CARD_STATES.IN_PROGRESS}) : card
             ));
         setActiveCards([]);
     }
 
-    function openCard(id, options) {
+    function openCard(id:number, options) {
```

Comment:
Also prefer arrow functions
```suggestion
    function openCard(id: number, options) {
```